platform/core/src/poly_out.rs

```diff
@@ -0,0 +1,112 @@
+use std::fs::DirBuilder;
+use std::path::Path;
+use uuid::Uuid;
+
+type ResourceId = String;
+type FsId = String;
+
+struct Metadata {}
+struct Content {}
+
+trait PolyOutTrait {
+    fn metadata(resource_id: ResourceId) -> Result<Metadata, String>;
```

Comment:
Since you said these are still `polypod:` URLs representing actual paths, I don't think we should call them "IDs" - this comment also applies to all the other calls that get a URL/path as input - i.e. most of them.

Also, I prefer the name `stat` - it's quite common POSIX terminology for this functionality. It's short for `status`, if we want to write it out.